pluginops/prs.go

```diff
@@ -0,0 +1,249 @@
+			if err != nil {
+				return err
+			}
+			if !ok {
+				continue
+			}
+
+			err = mergePR(context.Background(), client, pr)
+			if err != nil {
+				return err
+			}
+
+			fmt.Printf("Merged %s\n\n", pr.GetHTMLURL())
+		}
+
+		return nil
+	},
+}
+
+type prInfo struct {
+	Repository string
+	*github.PullRequest
+	reviews []*github.PullRequestReview
+}
+
+func getPluginPRs(ctx context.Context, client *github.Client) ([]*prInfo, error) {
+	user, _, err := client.Users.Get(ctx, "")
+	if err != nil {
+		return nil, err
+	}
+
+	query := fmt.Sprintf("is:pr is:open author:%v org:%v", user.GetLogin(), org)
+
+	issues, _, err := client.Search.Issues(ctx, query, &github.SearchOptions{}) // TODO: handle pagination
+	if err != nil {
+		return nil, err
+	}
+
+	var prs []*prInfo
+
+	var wg sync.WaitGroup
+	for _, issue := range issues.Issues {
+		if issue.IsPullRequest() {
+			url := issue.GetHTMLURL()
+			url = strings.TrimPrefix(url, "https://github.com/")
+			url = strings.TrimPrefix(url, org+"/")
+
+			repo := strings.Split(url, "/")[0]
+
+			if !isPluginRepo(repo) {
+				continue
+			}
+
+			wg.Add(1)
+			go func(repo string, number int) {
+				defer wg.Done()
+				pr, _, err := client.PullRequests.Get(ctx, org, repo, number)
+				if err != nil {
+					log.WithError(err).Warn("failed to get PR")
+				}
+
+				reviews, _, err := client.PullRequests.ListReviews(ctx, org, repo, number, &github.ListOptions{}) // TODO: handle pagination
+				if err != nil {
+					log.WithError(err).Warn("failed to get PR")
+				}
+				prs = append(prs, &prInfo{
+					Repository:  repo,
+					PullRequest: pr,
+					reviews:     reviews,
+				})
+
+			}(repo, issue.GetNumber())
+
+		}
+	}
+
+	wg.Wait()
+
+	return prs, nil
+}
+
+func (pr *prInfo) readyToBeMerged() bool {
+	for _, l := range pr.Labels {
+		if l.GetName() == "Do Not Merge" {
+			return false
+		}
+		if l.GetName() == "Work In Progress" {
+			return false
+		}
+		if l.GetName() == "3: QA Review" {
+			return false
+		}
+	}
+
+	if pr.GetMergeableState() != "clean" {
+		return false
+	}
+
+	var hasOneApprovedReview bool
+	for _, r := range pr.reviews {
+		if r.GetState() == "APPROVED" {
+			hasOneApprovedReview = true
+		}
+		if r.GetState() == "CHANGES_REQUESTED" {
+			return false
+		}
+
+		if r.GetState() == "COMMENTED" {
+			return false
+		}
+
+		// If a review from dylan is pending, the PR is not ready to be merged.
+		if r.User.GetName() == "dylan.haussermann" && r.GetState() == "PENDING" {
+			return false
+		}
+	}
+	if hasOneApprovedReview {
+		return true
+	}
+
+	return true
+}
+
+func isPluginRepo(repoName string) bool {
+	if strings.HasPrefix(repoName, "mattermost-plugin-") {
+		return true
+	}
+
+	for k := range defaultMapping {
+		if k == repoName {
+			return true
+		}
+	}
+
+	return false
+}
+
+func mergePR(ctx context.Context, client *github.Client, pr *prInfo) error {
+	_, err := client.Issues.RemoveLabelForIssue(ctx, org, pr.Repository, pr.GetNumber(), "2: Dev Review")
```

Comment:
Does this throw an error if the label isn't applied?

Reply:
It does indeed. Updated the code to fix that case.